### ISSUE TITLE
Cop to check for .gitignore'd files

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#9247]
There is a [note regarding `Gemfile.lock`](https://rails.rubystyle.guide/#gemfile-lock), but it is under "Rails Style Guide". I think it is not only rails specific.

Imo, we could at least check for
1. `Gemfile.lock`
2. `db/schema.db` (supporting multi databases)
3. `db/structure.sql` (supporting multi databases)
4. some other examples?
TO NOT BE ignored.

We can check for some files TO BE ignored:
1. like `.env` related files
2. rails credentials related files
3. something else?

Make those lists configurable.

**Comment by marcandre** [comment]
> Make those lists configurable

Yes :-) For example `Gemfile.lock` is no longer officially recommended by rubygems in gitignore and it isn't anymore by default for `bundle gem my_new_gem` for example. I disagree with this, but a cop must be configurable for this. Autocorrection would be cool; I have to manually add it when I create a gem (which is often these days, playing with Ractor).

I'm also adding `.rubocop-http*` until #9181 is resolved.

**Comment by bbatsov** [comment]
Closing as stale — checking for .gitignore'd files is tangential to RuboCop's core mission as a Ruby linter. No engagement in 5+ years.
